Allocator Slaves Sorters Fix

djdhm opened the pull request:
Fixed slaves sorters and Hierarchical Allocator 
These changes are made to reduce resources fragmentation in mesos Cluster, this is a continuation of this [Pull Request](https://github.com/criteo-forks/mesos/pull/29) #see this [Confluence page](https://confluence.criteois.com/display/~d.dahmane/Mesos+Fragmentation+Internship) for more informations.
The working slaves sorters are : 
1- Lexicographic sorter 
2- CPU First sorter 
Changed the offers hashmap into an ordered map to send offers in the order set by the slaves sorter.

djdhm committed:
Remove the slaves shuffle

djdhm committed:
Removed random shuffle of slaves list

djdhm committed:
Fixed Lexigocraphic Sorter compare function

djdhm committed:
Fixing the cpu first slaves sorter

djdhm committed:
Added tsl library to dependencies

djdhm committed:
changed slaves resources map to an ordered map

kamaradclimber commented on src/master/allocator/allocator.cpp:
```diff
@@ -59,6 +59,9 @@ Try<Allocator*> Allocator::create(
   //
   // We also look for "HierarchicalDRF" since that was the
   // previous value for `DEFAULT_ALLOCATOR`.
+  LOG(INFO) << " Creation de l'allocateur";
```
I think you should remove these :)

djdhm replied:
Done

djdhm committed:
Removed unecessary logs

djdhm committed:
Making random sorter the default slave sorter

djdhm committed:
Added constructor of orderedmap from hashmap

djdhm committed:
Added sum function for ordered_map

djdhm committed:
Added orderedmap to tests

djdhm committed:
Fixed Tests

djdhm committed:
Run tests on Ubuntu-18.04 instead of Ubuntu:latest

The failing tests are maybe related to the change of version in running container of the wrokflow
